src/main/java/RPN.java

```diff
@@ -0,0 +1,130 @@
+import java.util.Stack;
+
+public class RPN
+{
+    private final Stack<Double> rpnStack;
+
+    public RPN ()
+    {
+        this.rpnStack = new Stack<>();
+    }
+
+    /**
+     * A method to evaluate a string, presumably an RPN (Reverse Polish Notation) expression
+     * Explanation of RPN: https://en.wikipedia.org/wiki/Reverse_Polish_notation
+     * @param userInput the string typed by the user, presumably an RPN expression
+     * @return a string, either the answer or an appropriate error message
+     */
+    public String evaluate(String userInput)
+    {
+        String answer = "Ended with error";
+        boolean hasError = false;
+
+        try
+        {
+            String[] splitExpression = userInput.split("\\s+");
+
+            for (String element : splitExpression)
+            {
+                if (isNumber(element))
+                {
+                    rpnStack.add(Double.parseDouble(element));
+                }
+                else if (isOperator(element))
+                {
+                    try
+                    {
+                        Double secondNumber = rpnStack.pop();
+                        Double firstNumber = rpnStack.pop();
+                        doOperation(rpnStack, element, firstNumber, secondNumber);
+                    }
+                    catch (Exception e)
+                    {
+                        answer = "Error: syntax error encountered";
+                        hasError = true;
+                        break;
+                    }
+                }
+                else    // token is neither a double nor an operator
+                {
+                    answer = "Error: invalid token encountered.";
+                    hasError = true;
+                    break;
+                }
+            }
+
+            if (!hasError)
+            {
+                answer = rpnStack.pop().toString();
+            }
+
+            if (!rpnStack.isEmpty())
+            {
+                answer = "Error: stack is too big. The expression is likely malformed.";
```

Comment:
RPN object can be reused.
What happens if it's malformed and then you try using it again.